pkg/gce-pd-csi-driver/controller.go

```diff
@@ -1828,10 +1830,22 @@ func (b *csiErrorBackoff) blocking(id csiErrorBackoffId) bool {
 	return blk
 }
 
-func (b *csiErrorBackoff) next(id csiErrorBackoffId) {
+func (b *csiErrorBackoff) code(id csiErrorBackoffId) codes.Code {
+	if code, ok := b.errorCodes[id]; ok {
+		return code
+	}
+	// If we haven't recorded a code, return unavailable, which signals a problem with the driver
```

Comment:
My comment was that your  asssumption here is wrong. `next()` will always record whatever error code it is passed, which is the error code returned from CodeForError. CodeForError ALWAYS returns an error code, so we will never default to this case here where you return Unavailable. From your comment, it seems like you want to return Unavailable when the error is not a User error, so if you want to do that then, you should check if code from code := errorCodes[id], is Internal, and if it is return Unavailable. Or change your comment to say  we will never hit this case and remove the line that says " (ie, it's not filtered out as a user problem)."

Reply:
I was concerned with what to do if err is nil. That's not the case for the current use of next(), so it's kind of theoretical (just as a nil CodeForError was back when it was a pointer).

Reply:
oh, nvm, CodeForError is now reasonable if err is nil.

Reply:
Yes you're right, I've updated the comment